tests/v1/e2e/test_context_length.py

```diff
@@ -0,0 +1,123 @@
+# SPDX-License-Identifier: Apache-2.0
+# SPDX-FileCopyrightText: Copyright contributors to the vLLM project
+"""
+end-to-end tests for context length corner cases of vLLM v1 model runner
+versus HuggingFace's transformers.
+
+This test verifies the following behavior: allow a prefill that fills the 
+model's maximum context length and then request a single new token. 
+
+Test strategy
+- Build a textual prompt that tokenizes to exactly ``max_model_len`` tokens.
+- Run vLLM generation requesting a single new token (max_tokens=1).
+- Run HF generation on the same prompt requesting a single token too. 
+- Assert both return the same number of generated tokens and the same ids.
+
+"""
+
+import pytest
+import torch
+from transformers import AutoModelForCausalLM, AutoTokenizer
+
+from vllm import LLM, SamplingParams
+from vllm.inputs import TokensPrompt
+
+
+@pytest.mark.parametrize("model", ["JackFram/llama-160m"])
+@pytest.mark.parametrize("max_model_len", [2048])
+@pytest.mark.parametrize("max_tokens", [1])
+def test_models(
+    monkeypatch: pytest.MonkeyPatch,
+    model: str,
+    max_model_len: int,
+    max_tokens: int,
+) -> None:
+    """Compare vLLM and HuggingFace when the prompt already fills the
+    model's maximum context length and we request a single new token.
+
+    The test ensures vLLM does not raise the "Sampled token IDs exceed the
+    max model length" assertion and that both vLLM and HF produce the same
+    single token when given the same inputs.
+    """
+
+    with monkeypatch.context() as m:
+        m.setenv("VLLM_USE_V1", "1")
+
+        # Construct a deterministic textual prompt that repeats a short token
+        # sequence ("1 2 3 ... 9 ") until the resulting tokenized sequence has
+        # length max_model_len.
+        token_block = " ".join(str(i) for i in range(1, 10)) + " "
+        repeats = max_model_len // len(token_block) + 1
+        # accounting for BOS and EOS tokens
+        long_text = (token_block * repeats)[:max_model_len - 2]
+        prompts = [long_text]
+
+        # Tokenize with HF tokenizer and verify we produced the expected number
+        # of tokens.
+        tokenizer = AutoTokenizer.from_pretrained(model)
+        prompt_ids = tokenizer(prompts)["input_ids"]
+        len_prompt_ids = len(prompt_ids[0])
+        assert len_prompt_ids == max_model_len, (
+            f"Tokenized prompt length ({len_prompt_ids}) != "
+            f"max model len ({max_model_len})")
```

Comment:
Why do you need to start from a string and tokenizer it? You can just create the `TokensPrompt` of the desired length directly? It would make the test a lot simpler.